lib/modules/submitHelper.js

```diff
@@ -51,41 +58,53 @@ const $repostWarning = _.once(() => $(string.html`
-			$(urlFieldDiv).parent().after($repostWarning());
-			urlField = ((urlFieldDiv.querySelector('#url'): any): HTMLInputElement);
-			srField = ((document.querySelector('#sr-autocomplete'): any): HTMLInputElement);
-			$([srField, urlField]).on('input keydown', _.debounce(updateRepostWarning, 300));
-			// No event is fired when reddit's js changes the subreddit field, so update whenever the user clicks
-			$('#suggested-reddits a, #sr-drop-down').on('click', updateRepostWarning);
-			// We would allow reddit to show/hide the message for link/text posts with #link-desc
-			// but some subreddits hide this box, so we'll do it manually.
-			const linkButton = document.querySelector('a.link-button');
-			const textButton = document.querySelector('a.text-button');
-			if (linkButton && textButton) {
-				linkButton.addEventListener('click', () => { updateRepostWarning(); });
-				textButton.addEventListener('click', () => { $repostWarning().hide(); });
-			}
-		}
+	implementOptions();
+	// Register submit CLI if not on the submit page
+	if (!isPageType('submit')) {
+		registerCommandLine();
 	}
+};
 
-	if (isPageType('submit') && module.options.uncheckSendRepliesToInbox.value) {
-		const sendReplies: ?HTMLInputElement = (document.querySelector('#sendreplies'): any);
-		if (sendReplies) {
-			sendReplies.checked = false;
+async function implementOptions() {
+	if (isPageType('submit')) {
+		// Implement submitHelper preferences if on the submit page
+		if (module.options.warnAlreadySubmitted.value) {
+			const urlFieldDiv = document.querySelector('#url-field');
+			if (urlFieldDiv) {
+				$(urlFieldDiv).parent().after($repostWarning());
+				urlField = ((urlFieldDiv.querySelector('#url'): any): HTMLInputElement);
+				srField = ((document.querySelector('#sr-autocomplete'): any): HTMLInputElement);
+				$([srField, urlField]).on('input keydown', _.debounce(updateRepostWarning, 300));
+				// No event is fired when reddit's js changes the subreddit field, so update whenever the user clicks
+				$('#suggested-reddits a, #sr-drop-down').on('click', updateRepostWarning);
+				// We would allow reddit to show/hide the message for link/text posts with #link-desc
+				// but some subreddits hide this box, so we'll do it manually.
+				const linkButton = document.querySelector('a.link-button');
+				const textButton = document.querySelector('a.text-button');
+				if (linkButton && textButton) {
+					linkButton.addEventListener('click', () => { updateRepostWarning(); });
+					textButton.addEventListener('click', () => { $repostWarning().hide(); });
+				}
+			}
 		}
-	}
 
-	if (isPageType('submit') && module.options.focusFormOnLoad.value) {
-		$('form.submit [name=url], form.submit [name=title]').filter(':visible').first().focus();
-	}
+		if (module.options.uncheckSendRepliesToInbox.value) {
+			const selector = isAppType('d2x') ? '[aria-labelledby="send-replies"]' : '#sendreplies';
+			try {
+				const sendReplies: ?HTMLElement = await findElement(selector);
```

Comment:
Also consider using `waitForDescendent`.

Reply:
@larsjohnsen You mean change the function name?

Reply:
Tried it. `waitForDescendants` adds another observer, which is causing the Send Reply Notifications toggle to switch whenever the React component for Submit is mounted. So, I think I'll stick to `findElement`, since it only fetches the element once.

Reply:
You might be confusing `waitForDescendant` and `watchForDescendants`. `waitForDescendant` will stop observing once it has found a match, but it may be a good idea to have it optionally time out.

Reply:
Okay, got it. My bad. Thanks!